### PR TITLE
chore: configure sandbox

### PR DESCRIPTION
**Pull request by ChaoticTempest**
Separated out the logic purely from https://github.com/near/workspaces-rs/pull/241. This allows configuring `config.json` so that sandbox can do extra stuff on top of what mainnet/testnet does such as being able to patch large states which resolves https://github.com/near/workspaces-rs/issues/223

### CODE REVIEW

File: workspaces/src/network/config.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+use serde_json::{Map, Value};
+
+use crate::error::ErrorKind;
+use crate::Result;
+
+/// Replaces all entries recursively from `from` into `into`.
+fn overwrite_map(from: Map<String, Value>, mut into: Map<String, Value>) -> Result<Value> {
+    for (key, from_val) in from.into_iter() {
+        let Some(into_val) = into.remove(&key) else {
```
Comment (DavidM-D) [inline workspaces/src/network/config.rs:25]: This is unstable rust.
Do we require unstable to compile this crate?

File: workspaces/src/network/config.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        // Default is 1GB which should suit most contract sizes.
+        Err(_err) => 1024 * 1024 * 1024,
+    };
+
+    Ok(max_files)
+}
+
+/// Set extra configs for the sandbox defined by workspaces.
+pub(crate) fn set_sandbox_configs(home_dir: &Path) -> Result<()> {
+    overwrite(
```
Comment (DavidM-D) [inline workspaces/src/network/config.rs:95]: Can this be replaced by [json_patch's merge](https://docs.rs/json-patch/0.2.7/json_patch/fn.merge.html) and a bit of IO?

**Review by DavidM-D — APPROVED** [review]
Logic looks good, but can we just replace most of this with this lib?

https://docs.rs/json-patch/0.2.7/json_patch/fn.merge.html

File: workspaces/src/network/config.rs (hunk 1 above)
Comment (itegulov) [reply workspaces/src/network/config.rs:25]: Are you talking about the destructing assignment? This was stabilized a year ago I think

**Comment by ChaoticTempest** [comment]
@DavidM-D didn't even know that library existed. At first was trying to find similar functionality inside `serde_json`

**Comment by DavidM-D** [comment]
Not 100% sure we need to test `json_patch`, but your call and LGTM

**Comment by ChaoticTempest** [comment]
I was keeping it just in case json_patch changes, but we're pinning the version anyways

**Pull request merged**